apprise/Apprise.py

```diff
@@ -344,13 +451,8 @@ def notify(self, body, title='', notify_type=NotifyType.INFO,
 
         # Prepare attachments if required
         if attach is not None and not isinstance(attach, AppriseAttachment):
-            try:
-                attach = AppriseAttachment(
-                    attach, asset=self.asset, location=self.location)
-
-            except TypeError:
-                # bad attachments
-                return False
+            attach = AppriseAttachment(
```

Comment:
Just curious why you're allowing `TypeError` to raise here?

Reply:
So with this _notifyall() function, there are three cases to tease out from the outcome:

1.The configuration was valid, and notifications were sent: The generator yields one or more items. This tells notify() to return True or False.
2. The configuration was valid, but no notifications were sent: The generator yields no items. This tells notify() to return None.
3. The configuration was invalid, but it's possible some notifications were sent. The generator must yield the corresponding items *and* tell notify() there was an error condition, so that it can return False.

Taking a cue from the existing code, I promoted `TypeError` to distinguish the third case. Both versions of notify() catch the exception and thereby know to return False.

One alternative would be to create a sentinel value that the generator could yield to signal a failure, but using an exception for flow control is more Pythonic, IIRC.

Reply:
Okay, so, upon further review, it turns out it's not possible to send *any* notifications with a bad `server.notify_format` value. So the "some notifications" possibility I was worried about in case 3 is actually nothing to be concerned about.

My point about using exceptions to notify the caller of a configuration failure, however, still stands.